### PR TITLE
API-15999 - add fhirVersion, version, label fields to APIVersionModel

### PR DESCRIPTION
**Pull request by christineyost**
This PR is for [API-15999](https://vajira.max.gov/browse/API-15999). It adds fhirVersion, version, and label fields to APIVersionModel in order to handle metadata for the Patient Health API. The new `version` field is calculated based on `majorVersion` and `fhirVersion`. The APIVerision unique index is now a combination of `api` and `version` fields, instead of `api` and `majorVersion`.

This change is documented in an ADR linked on the Jira ticket.

Other notable changes:
* `oasdiff.js` was reformatted by Prettier. The only significant change in that file is the 'v' was removed from `pagePattern`
* Updated the generated migration script to include setting the `version` field for existing APIVersions
* Added `doesAPIVersionExist` method to `APIVersionService` so the `APIVersionController` wouldn't need to know how to construct the `version` field

Other test changes:
* `apiVersion.filter.test.ts` - added test for `filterAPIVersionOASStatusResponse()` method
* `apiVersionresolver.middleware.test.ts` - added test for api not being set on the request
* `oas.service.integration.ts` - added checks to resolve warnings/errors about variables possibly being null/undefined
* `oas.service.test.ts` - added a couple tests for edge-case errors being thrown
* `portalStore.service.test.ts` - added test for `uploadOASDiff` method

### CODE REVIEW

File: src/differ/differ.service.ts [hunk 1]
```diff
@@ -35,7 +35,7 @@ export class DifferService {
       }`;
     }
 
-    const titleText = `Comparison of API ${name} ver.${version}${suffix}`;
+    const titleText = `Comparison of API ${name} ${version}${suffix}`;
```
Comment (mheusgit) [inline src/differ/differ.service.ts:38]: Good catch

**Comment by mheusgit** [comment]
Before PR is merged it requires some coordination with work in API-22131

File: src/oas/transformer/oas.servers.transformer.ts [hunk 2]
```diff
@@ -38,9 +38,9 @@ class OASServersTransformer {
 
   private createServersElement(
     exposedRoute: string,
-    apiVersion: number,
+    majorVersion: number,
   ): Array<OASServer> {
-    const version = `v${apiVersion}`;
+    const version = `v${majorVersion}`;
```
Comment (dsteeve) [inline src/oas/transformer/oas.servers.transformer.ts:43]: sad that this 'v' still exists, but works for now I think. maybe put a comment why this couldn't just use "version" (because of fhir versions would mess it up - but someone might forget why later on)

**Review by dsteeve — APPROVED** [review]
Lots of code to look at! 

I have a few minor nits, no show stoppers.  I did bring down and add data, run the migrations and ran through several items.  All worked and looked great! 
Please maybe fix the minor nits before merging?

File: README.md [hunk 3]
```diff
@@ -500,7 +500,9 @@ If you are adding a new model make sure to update the entities list with your ne
...
 | loast_config_id  | Integer  | No | Foreign key to LOAST_CONFIG(ID) | - | Unique ID for LOAST configuration for this API version |
 | internal_only  | boolean  | Yes | true or false | - | Flag used to determine if API allows access outside of VA |
-| major_version  | Integer  | Yes | - | - | Denotes version number.  major_version and api_id values must be unique. Combined with "v" for OAS path generation and to detect & remove version numbers from OAS forward paths (Ex: /address/`v1`/candidate).  Also used for hash generation. |
+| major_version  | Integer  | Yes | - | - | Denotes the major version number. Used during OAS transformation, when generating the server urls |
+| fhir_version  | Varchar (255)  | No | - | "r4" | Denotes the fhir version. This field is for use of APIVersions that belong to a FHIR API. |
+| version  | Varchar (255)  | Yes | v{major_version}-{fhir_version} | "v0-argonaut" | This field is not specified on creation or update, but is calculated based on major_version and fhir_version. The combination of api_id and version values must be unique. Used for OAS path generation and OAS hash generation |
```
[the remark this answers is not in the document]
Comment (christineyost) [reply README.md:505]: Yes! Added

File: src/oas/transformer/oas.servers.transformer.ts (hunk 2 above)
Comment (christineyost) [reply src/oas/transformer/oas.servers.transformer.ts:43]: Yes - maybe it would be better to move this logic to APIVersionModel as well - like generateOASServerVersion or something? We could make another ticket to refactor this

Comment (christineyost) [reply src/oas/transformer/oas.servers.transformer.ts:43]: Added a comment for now

File: src/api/version/apiVersion.model.ts [hunk 4]
```diff
@@ -245,6 +270,13 @@ class APIVersionModel extends BaseEntity {
     this.securityModel = BaseSecurity.constructSecurity(this.securityType);
     return this;
   }
+
+  static generateVersion(
+    majorVersion: number,
+    fhirVersion: string | undefined,
+  ): string {
+    return `v${majorVersion}${fhirVersion ? `-${fhirVersion}` : ''}`;
+  }
```
[the remark this answers is not in the document]
Comment (christineyost) [reply src/api/version/apiVersion.model.ts:279]: I think it was being tested through the constructor and update tests, but I do think it's more clear to add a test just for this method - added!
